storage/src/vespa/storage/storageserver/documentapiconverter.cpp

```diff
@@ -404,4 +404,14 @@ DocumentApiConverter::transferReplyState(api::StorageReply& fromMsg, mbus::Reply
     }
 }
 
+std::shared_ptr<const BucketResolver> DocumentApiConverter::bucketResolver() const {
+    std::lock_guard lock(_mutex);
+    return _bucketResolver;
+}
+
+void DocumentApiConverter::setBucketResolver(std::shared_ptr<const BucketResolver> resolver) {
+    std::lock_guard lock(_mutex);
```

Comment:
PS, with sweet C++17 class template argument deduction swag, we don't have to write `std::lock_guard<std::mutex>` anymore 🎅👍